packages/server/src/api/controllers/auth.ts

```diff
@@ -24,7 +25,7 @@ export async function fetchSelf(ctx: any) {
   }
 
   const appId = context.getAppId()
-  const user = await getFullUser(ctx, userId)
+  const user: ContextUser = await getFullUser(ctx, userId)
```

Comment:
Can we type the `getFullUser` fn to also return `ContextUser` to make sure these line up. 

We should probably better define the forms the user can take, that's a lot of different concepts in play like
- user built by app-service middleware
- user built by worker-service middleware
- user cached in redis
- db user
- app user
- /api/self user
- /api/global/self user

I'd be up for doing a session where we outline the properties used in each case, make sure we've got the correct types in place and plan for where they'll be used.

Reply:
Yeap, I think handling this as a separate PR and defining it all is a great idea - I have noticed this is a bit of an issue in a few places!